### ISSUE TITLE
batchnorm in federated learning

### ISSUE THREAD

**Issue by hwan-sig** [debcaldarola/fedsam#8]
Hello, how did you average the weight of batch normalization?
When the server averages local models, did you average all the parameters including batch norm, and distribute the weight of batch norm to clients?

Thank you

**Comment by debcaldarola** [comment]
Hi,
yes the BN parameters are averaged together with the rest of the model, and then distributed to the clients as part of the model (we do not distribute BN statistics separately to the clients). This happens in this function https://github.com/debcaldarola/fedsam/blob/2500c29e877806ecca19960a8be242f520092e58/models/servers/fedavg_server.py#L133.

**Comment by debcaldarola** [comment]
yes exactly! please note that the models in this repo do not contain BN layers
